internal/testutils/tls_gen.go

```diff
@@ -170,9 +178,17 @@ func (c *CertBuilder) clientCertTemplate() *x509.Certificate {
 func (c *CertBuilder) serverCertTemplate() *x509.Certificate {
 	return &x509.Certificate{
 		SerialNumber:          big.NewInt(3),
-		Subject:               pkix.Name{CommonName: "server.com"},
+		Subject:               pkix.Name{CommonName: fmt.Sprintf("server-%s", c.commonName)},
 		NotBefore:             time.Now(),
 		NotAfter:              time.Now().Add(time.Hour),
 		BasicConstraintsValid: true,
 	}
 }
+
+func BuildInvalidKeyPair(caCertPem, clientCertPem, nonMatchingClientKeyPem bytes.Buffer) *ClientCerts {
```

Comment:
This function takes 3 arguments and returns a struct filled with those 3 arguments. Do we really need it?